### PR TITLE
Extract GuidedTour-related components from A4A

### PR DESCRIPTION
**Pull request by okmttdhr**
Related to 
- https://github.com/Automattic/dotcom-forge/issues/10138
- https://github.com/Automattic/wp-calypso/pull/97542
- https://github.com/Automattic/wp-calypso/pull/91100

## Proposed Changes

This PR extracts `GuidedTour`-related components from A4A and creates shared ones. 

## Why are these changes being made?

- https://github.com/Automattic/dotcom-forge/issues/10138
- Please refer to the Project Thread pbxlJb-6J5-p2 and the proposal post pfYzsZ-Y4-p2 for more context.

## Testing Instructions

Prerequisite 

* Go to Developer Console 
* POST to /me/preferences with the following data in `calypso_preferences`

```
{
  "site-management-panel": {"dismiss": false},
  "a4a-sites-add-new-site-tour-step-1": {"dismiss": false},
  "a4a-sites-add-new-site-tour-step-2": {"dismiss": false},
  "a4a-sites-tour": {"dismiss": false},
  "a4a-marketplace-tour": {"dismiss": false},
  "a4a-boost-agency-visibility-tour": {"dismiss": false},
  "a4a-start-referrals": {"dismiss": false},
  "a4a-invite-team": {"dismiss": false}
}
```

Screenshot 2024-12-19 at 13 00 01

Calypso Live

* Go to /sites
* Select any site
* Observe the tour starts correctly 
* Refresh the page after the tour ends
* Observe the tour does not restart

A4A Live

* Go to /sites?tour=sites-walkthrough
	* Observe the tour starts correctly 
	* Refresh the page after the tour ends
	* Observe the tour does not restart
* Go to /sites?tour=add-new-site
	* Observe the tour starts correctly 
	* Refresh the page after the tour ends
	* Observe the tour does not restart

CC: @Automattic/a4a-genesis 

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [x] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Review by fushar — APPROVED** [review]
✅ Calypso /sites

image

- Observe the tour starts correctly ✅
- Refresh the page after the tour ends ✅
- Observe the tour does not restart ✅

✅ A4A /sites?tour=sites-walkthrough

image

- Observe the tour starts correctly ✅
- Refresh the page after the tour ends ✅
- Observe the tour does not restart ✅


✅ A4A /sites?tour=add-new-site

image

- Observe the tour starts correctly ✅
- Refresh the page after the tour ends ✅
- Observe the tour does not restart ✅

File: client/components/guided-tour/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Guided Tour Components
```
Comment (fushar) [inline client/components/guided-tour/README.md:1]: Nice 👍

**Comment by cleacos** [comment]
Tested it through A4A, and it works well 👍 

> # Guided Tour Components

Nice 🙂 

Thank you!

**Comment by cleacos** [comment]
@okmttdhr  I noticed an issue, but I'm unsure if it was before this PR as a regression issue. I don't know if @fushar noticed the same. When I go through the steps (Sites tour), and when I go to the next step, it clicks on the previous column header and it opens the "order" selector:

image

image

**Comment by okmttdhr** [comment]
> @okmttdhr I noticed an issue, but I'm unsure if it was before this PR as a regression issue. I don't know if @fushar noticed the same. When I go through the steps (Sites tour), and when I go to the next step, it clicks on the previous column header and it opens the "column order" selector:

This behavior existed before this PR, and I assumed it was expected. I should've mentioned this. 
If it's all right, I can follow up on this next Monday!

**Comment by cleacos** [comment]
Adding @madebynoam and @jeffgolenski to get some feedback about that "possible" issue.

**Comment by jeffgolenski** [comment]
Thanks @cleacos - yeah that's definitely unexpected behavior. Good find and thanks for diving in @okmttdhr !

**Pull request merged**